### DISCUSSION TITLE
Qi charging heat issues

### DISCUSSION THREAD

**Discussion by machacekch** [bitcraze/discussions#1435] [Q&A]
I'm using a cf2 with a Qi + AI + LH deck setup and the 350mAh battery.
After about 5 minutes of wireless charging `pm.state` jumps to `2` ("charged") and the blue LED stops blinking even though the battery isn't full yet (e.g. `pm.batteryLevel` is still at 10%).
Following this, `pm.state` seems to oscillate between `0` and `2`. At that point the Qi deck is rather hot to the touch. Blowing at the cf2 makes it resume charging, so it seems to be some kind of overheat protection.

My questions:
* Is that overheat protection software-based and if so, where is it implemented?
* Is it the battery or the Qi deck that's overheating? (Or the charging circuit on the cf2 itself?)
* Is this to be expected? I should say that I have the AI deck mounted below the main board, so the Qi deck doesn't have much airflow around it.
* Does the choice of battery (250mAh vs. 350mAh) have anything to do with it?

**Comment by machacekch** [comment]
I should add that charging via USB works just fine.

**Comment by tobbeanton** [comment]
> Is that overheat protection software-based and if so, where is it implemented?

You can turn it off by building the nRF firmware with the define [DISABLE_CHARGE_TEMP_CONTROL](https://github.com/bitcraze/crazyflie2-nrf-firmware/blob/master/src/pm.c#L424C9-L424C36)

> Is it the battery or the Qi deck that's overheating? (Or the charging circuit on the cf2 itself?)

It is the cf2 that checks the temperature and can be turned off. Another solution is to spin the propellers slowly to cool it down.

> Is this to be expected? I should say that I have the AI deck mounted below the main board, so the Qi deck doesn't have much airflow around it.

It depends on the ambient temperature, the charger, distance from Qi-deck to cf2, etc.

> Does the choice of battery (250mAh vs. 350mAh) have anything to do with it?

This shouldn't effect it.
